### DISCUSSION TITLE
Add alternate names/tags to function docblocks for the LSP to pickup / people to find by find-all

### DISCUSSION THREAD

**Discussion by inoas** [gleam-lang/gleam#1645] [Ideas & suggestions]
See https://github.com/gleam-lang/stdlib/pull/307#issuecomment-1133779774

If above gets merged or not does not matter for the sake of this argument:

Users onboarding from other languages may be used to certain nomenclature.
For instance it took me a while to find a `map()` function. People may be looking for `abs()` etc. 

Would it be nice if when typing those functions the LSP could scan the docblock for a tag of alternate names?

Tagged @NicklasXYZ

**Comment by lpil** [comment]
I like this idea a lot. We should also think about what the syntax might be like

**Reply by inoas** [reply]
Some ideas:

```gleam
/// @aliases: abs, absolute
/// @aliases: map

/// @also: abs, absolute
/// @also: map

/// @also_named: abs, absolute
/// @also_named: map

/// @known_as: abs, absolute
/// @known_as: map
```

**Reply by lpil** [reply]
I would not want any structured information to live within comments. I think the comments should be meaningless to the compiler and not be parsed, it should instead operate on Gleam syntax

**Reply by inoas** [reply]
```gleam
/// Doc block here
///
@alt_names: abs, absolute
pub fn absolute_value(number x:Int) -> Int {
  todo
}
```

```gleam
/// I don’t like this
///
pub fn absolute_value/absolute/abs(number x:Int) -> Int {
  todo
}
```

```gleam
/// Neither this really
///
pub fn absolute_value(number x:Int) -> Int {
  todo
}, abs, absolute
```

**Reply by lpil** [reply]
That first one looks nicest to me. I imagine we could have other attributes too such as ones for depredation

**Reply by inoas** [reply]
Deprecation? Yes <3

**Reply by inoas** [reply]
```
@deprecated
@deprecated_use: foo
```

**Reply by inoas** [reply]
Ref: https://github.com/gleam-lang/stdlib/pull/302#discussion_r880896828
We could also have `@also_see module.function` if thats beneficial enough.

**Comment by inoas-nbw** [comment]
```rust
@aliases to_int
pub fn truncate(...
```
```rust
@aliases abs
pub fn absolute_value(...
```
```rust
@alias count
pub fn length(...
```
```rust
@aliases distinct
pub fn unique(...
```

**Comment by inoas** [comment]
Could include an optional description why that alias exists.
See for example: https://github.com/gleam-lang/stdlib/issues/599#issuecomment-2123972776
